Nanoleaf-wpf/Views/Layout/PercentageProfileWindow.xaml.cs

```diff
@@ -19,13 +12,107 @@ namespace Winleafs.Wpf.Views.Layout
     /// </summary>
     public partial class PercentageProfileWindow : Window
     {
+        private PercentageProfile _profile;
+
         public PercentageProfileWindow()
         {
             InitializeComponent();
 
             LayoutDisplay.SetWithAndHeight((int)LayoutDisplay.Width, (int)LayoutDisplay.Height);
             LayoutDisplay.DrawLayout();
             LayoutDisplay.EnableClick();
+            LayoutDisplay.DisableColorTimer();
+
+            if (UserSettings.Settings.ActiveDevice.PercentageProfile != null)
+            {
+                var serialized = JsonConvert.SerializeObject(UserSettings.Settings.ActiveDevice.PercentageProfile); //Deep copy the profile when editing
+                _profile = JsonConvert.DeserializeObject<PercentageProfile>(serialized);
+
+                BuildStepList();
+
+                foreach (var step in _profile.Steps)
+                {
+                    LayoutDisplay.LockPanels(step.PanelIds);
+                }
+            }
+            else
+            {
+                _profile = new PercentageProfile();
+            }
+        }
+
+        private void Plus_Click(object sender, RoutedEventArgs e)
+        {
+            if (LayoutDisplay.SelectedPanelIds.Count > 0)
```

Comment:
In general, it would be best practise to revert this condition and just `return;` if its 0.
This reduces nesting, within this function it is not necessary so I will leave it to you.